### ISSUE TITLE
WASM: The inner handler has not been assigned - Configure AuthorizationMessageHandler

### ISSUE THREAD

**Issue by paulguz** [dotnet/AspNetCore.Docs#18778]
When trying to use the code in the section #configure-authorizationmessagehandler, I receive the 

> error: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
>       Unhandled exception rendering component: The inner handler has not been assigned.

In a vanilla Blazor WASM app (from the template) have the following in Program.cs:

```
	builder.Services.AddTransient(sp =>
			{
				return new HttpClient(sp.GetRequiredService<AuthorizationMessageHandler>()
						.ConfigureHandler(
								authorizedUrls: new[] { $"https://api.mysite.com" },
								scopes: new[] { "scope1", "scope2" }))

				{
					BaseAddress = new Uri($"https://api.mysite.com")
				};
			});
```
I then call Http.GetFromJsonAsync and the above error occurs.

If I configure a named client (i.e. use AddHttpClient rather than AddTransient) otherwise identically, I don't get the error.

Is there something missing in the docs, or is it something I'm doing wrong?

---
#### Document details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: f2cdb52c-843f-5944-429d-11ea821c1e22
* Version Independent ID: 9c4f99ef-f070-b527-4efc-96a04798d9ef
* Content: [ASP.NET Core Blazor WebAssembly additional security scenarios](https://docs.microsoft.com/en-gb/aspnet/core/security/blazor/webassembly/additional-scenarios?view=aspnetcore-3.1#configure-authorizationmessagehandler)
* Content Source: [aspnetcore/security/blazor/webassembly/additional-scenarios.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/security/blazor/webassembly/additional-scenarios.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Hello @paulguz ... Open that for engineering at ...

https://github.com/dotnet/aspnetcore/issues

... add a `cc: @guardrex` to your opening comment.

We can leave this open here while Javier responds to your engineering issue.

**Comment by guardrex** [comment]
@paulguz You changed `builder.HostEnvironment.BaseAddress` to a string URI. Are they the same value?

**Comment by paulguz** [comment]
No, this is a standalone WASM app that is communicating with an existing API elsewhere.

**Comment by guardrex** [comment]
Why is the base address the address of the API? Did you try ...

```csharp
BaseAddress = new Uri(builder.HostEnvironment.BaseAddress)
```

... and it also failed the same way?

**Comment by paulguz** [comment]
@guardex Yes, it also fails when I tried that.  It works if I set it up as a named client.  

> Why is the base address the address of the API?
The Base Address of the Http Client **should** be the address of the api, shouldn't it? It is "the base adress ... of the internet resource used when sending requests".

**Comment by guardrex** [comment]
Oh, sorry ... yes. I forgot it's the host env base address in the example because this is a **_hosted Blazor solution_** that the example is showing. It just threw me off. *MORE* ☕☕☕☕☕ *PLZ!* :smile:

It's unrelated to your issue, but I don't think I care for that example ... the URL of `https://www.example.com/base` and `BaseAddress` that would be `https://www.example.com/base`. Seems like we could explain that out a bit more or present a clearer example.

**Comment by guardrex** [comment]
> That code snippet is wrong. Likely because you are not using HttpClientFactory. In that case you need to create an HttpClientHandler instance and assign it as the innerhandler of the AuthorizationMessageHandler manually.

Strange. I'll research how the bad code came in and get it fixed immediately.

**Comment by guardrex** [comment]
It came in from the Preview 5 blog post at https://devblogs.microsoft.com/aspnet/blazor-webassembly-3-2-0-preview-5-release-now-available/. It's the example code provided at the end of the Attach tokens to outgoing requests section, except that we changed the registration from AddSingleton to AddTransient later.

It might have gone stale at some point. It wasn't caught on dotnet/AspNetCore.Docs#18616 when I made a minor change to it while adding the Custom AuthorizationMessageHandler class coverage.

**Comment by guardrex** [comment]
Closing as fixed.
